networkit/cpp/scd/LocalT.cpp

```diff
@@ -0,0 +1,191 @@
+        }
+
+        assert(debug_external_triangles == external_triangles);
+        assert(debug_internal_triangles / 3 == internal_triangles);
+
+        for (node ls : shell) {
+            count s_external = 0, s_internal = 0, s_semi_internal = 0;
+            local_graph.forTrianglesOf(local_graph.toGlobal(ls),
+                                       [&](node lv, node lw, edgeweight, edgeweight, edgeweight) {
+                                           if (in_result[lv] && in_result[lw]) {
+                                               ++s_internal;
+                                           } else if (in_result[lv] || in_result[lw]) {
+                                               ++s_semi_internal;
+                                           } else {
+                                               ++s_external;
+                                           }
+                                       });
+
+            assert(s_semi_internal == node_semi_internal_triangles[ls]);
+            assert(s_internal == node_internal_triangles[ls]);
+            assert(s_external == node_external_triangles[ls]);
+        }
+#endif
+    };
+
+    // init community with seed set
+    for (node u : s) {
+        node lu = local_graph.ensureNodeExists(u);
+        result.insert(u);
+        in_result[lu] = true;
+        updateShell(u);
+    }
+
+    auto get_score = [](count int_triangles, count ext_triangles) -> count {
+        return std::max<int64_t>(
+            0, int_triangles
+                   * (static_cast<int64_t>(int_triangles) - static_cast<int64_t>(ext_triangles)));
+    };
+
+    // expand (main loop)
+    node uMax = none;
+    do {
+
+        uMax = none;
+        count best_score = get_score(internal_triangles, external_triangles);
+        count best_external_triangles = none;
+
+        for (node lv : shell) {
+            count new_internal_triangles = internal_triangles + node_internal_triangles[lv];
+            count new_external_triangles =
+                external_triangles + node_external_triangles[lv] - node_semi_internal_triangles[lv];
+
+            count new_score = get_score(new_internal_triangles, new_external_triangles);
+
+            if (new_score > best_score
+                || (new_score == best_score && new_external_triangles < best_external_triangles)) {
+                uMax = lv;
+                best_score = new_score;
+                best_external_triangles = new_external_triangles;
+            }
+        }
+
+        if (uMax != none) {
+            node gUMax = local_graph.toGlobal(uMax);
+            result.insert(gUMax);
+            shell.erase(uMax);
+            in_result[uMax] = true;
+            updateShell(gUMax);
+            assert(external_triangles == best_external_triangles);
+            assert(get_score(internal_triangles, external_triangles) == best_score);
+        }
+
+    } while (uMax != none);
+    return result;
+}
+
+} /* namespace NetworKit */
```

Comment:
In this file many variables use snake_case.

Reply:
I changed all variables to use lowerCamelCase.